### ISSUE TITLE
Functionality to support users with `modelr::data_grid` into `newdata`

### ISSUE THREAD

**Issue by athowes** [epinowcast/epidist#280]
The workflow (see #52) we have for creating `newdata` (that goes into `tidybayes`) is:

```r
newdata_sex <- obs_prep |>
  as.data.frame() |>
  modelr::data_grid(sex) |>
  mutate(obs_t = NA, pwindow_upr = NA, swindow_upr = NA)
```

Perhaps we want to support users to do this somehow?

Maybe just the mutation part? It could have options to specify e.g. "no censoring" or "no truncation" rather than having to know what to do yourself? (But also we should make it more obvious via the names.)

**Comment by seabbs** [comment]
I also would be surprised if that as.data.frame was actually needed?

**Comment by athowes** [comment]
I think it's needed currently but I don't think that it should be needed.

**Comment by athowes** [comment]
> I think ideally it would dispatch on the class of the model people have used

Yes I agree. Given that the parts that are required are the LHS of the formula which are model specific. It could be that in future we allow users to "predict without censoring" using a different model in a different way than by setting window variables to zero.
